reactor-core/src/jcstress/java/reactor/core/publisher/MonoDelayStressTest.java

```diff
@@ -1,175 +1,123 @@
 
 import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE_INTERESTING;
 
 public abstract class MonoDelayStressTest {
 
 	@JCStressTest
-	@Outcome(id = {"1, 1, 0, 0, 0"}, expect = ACCEPTABLE, desc = "No error dropped, composite error delivered. Cancel signal is late in that case")
-	@Outcome(id = {"1, 1, 1, 1, 0", "1, 1, 0, 1, 0"}, expect = ACCEPTABLE, desc = "Main error possibly dropped, inner or composite error delivered. Main is cancelled. Cancel signal is late")
-	@Outcome(id = {"1, 1, 1, 0, 1", "1, 1, 0, 0, 1"}, expect = ACCEPTABLE, desc = "Inner error possibly dropped, main or composite error delivered. Inner is cancelled. Cancel signal is late")
-	@Outcome(id = {"1, 0, 1, 1, 1", "1, 0, 2, 1, 1"}, expect = ACCEPTABLE, desc = "No error delivered. One composite error delivered or both errors are dropped. Cancel signal is propagated to both")
+	@Outcome(id = {"1, 0, true"}, expect = ACCEPTABLE, desc = "Request before tick was delivered")
+	@Outcome(id = {"1, 0, false"}, expect = ACCEPTABLE, desc = "Request AFTER tick was delivered")
 	@State
-	public static class InnerOnErrorAndOuterOnErrorAndCancelStressTest {
+	public static class RequestAndRunStressTest {
 
-		final StressSubscriber<Integer> subscriber = new StressSubscriber<Integer>(1L);
+		/*
+		Implementation notes: in this test we use the VirtualTimeScheduler to better coordinate
+		the triggering of the `run` method. We also use the hasProgressed AtomicBoolean to
+		track whenever the delayTrigger happens before the subscribe actor.
+		 */
 
-		StressSubscription<Integer> subscriptionOuter;
-		StressSubscription<Integer> subscriptionInner;
+		final StressSubscriber<Long> subscriber = new StressSubscriber<>(1L);
+		final VirtualTimeScheduler   virtualTimeScheduler;
+		final MonoDelay              monoDelay;
+		final AtomicBoolean          hasProgressed = new AtomicBoolean();
 
 		{
-			new Mono<Integer>() {
-				@Override
-				public void subscribe(CoreSubscriber<? super Integer> actual) {
-					subscriptionOuter = new StressSubscription<>(actual);
-					actual.onSubscribe(subscriptionOuter);
-					actual.onNext(1);
-				}
-			}
-			.delayUntil(__ -> new Mono<Integer>() {
-				@Override
-				public void subscribe(CoreSubscriber<? super Integer> actual) {
-					subscriptionInner = new StressSubscription<>(actual);
-					actual.onSubscribe(subscriptionInner);
-				}
-			})
-			.subscribe(subscriber);
+			virtualTimeScheduler = VirtualTimeScheduler.create();
+			monoDelay = new MonoDelay(1, TimeUnit.NANOSECONDS, virtualTimeScheduler);
+			monoDelay.subscribe(subscriber);
 		}
 
 		@Actor
-		public void errorOuter() {
-			subscriptionOuter.actual.onError(new RuntimeException("test1"));
-		}
-
-		@Actor
-		public void errorInner() {
-			subscriptionInner.actual.onError(new RuntimeException("test2"));
+		public void delayTrigger(IIZ_Result r) {
+			if (hasProgressed.compareAndSet(false, true)) {
```

Comment:
this is not reliable since a CAS and the real invocation of `advanceTimeBy` is unconnected and a compiler can change the total order of ops and reorder this operation with the request actor. That says that I would better rely on the internal state of the operator to observe the final state that has already reflect what happened rather than trying to assume it before really doing it